src/federation/index.ts

```diff
@@ -904,6 +906,61 @@ federation.setObjectDispatcher(
   },
 );
 
+federation.setObjectDispatcher(
+  Flag,
+  "/reports/{id}",
+  async (ctx, { id }) => {
```

Comment:
@dahlia okay, I think this gives us correct-ish dereferencing of Flag activities. The access control is a little wonky since it has to be from the targetAccount, not the remote instance actor (because we don't know who that is). Arguably, we could just check if the hosts match, but that could be potentially dangerous.

Though, this does make you one of the first servers I know to have dereferenceable Flag activities.